src/v/cloud_storage_clients/s3_client.cc

```diff
@@ -203,12 +203,18 @@ request_creator::make_list_objects_v2_request(
     if (prefix.has_value()) {
         key = fmt::format("{}&prefix={}", key, (*prefix)().string());
     }
-    if (delimiter.has_value()) {
-        key = fmt::format("{}&delimiter={}", key, *delimiter);
+    if (start_after.has_value()) {
+        key = fmt::format("{}&start-after={}", key, *start_after);
     }
     if (max_keys.has_value()) {
         key = fmt::format("{}&max-keys={}", key, *max_keys);
     }
+    if (continuation_token.has_value()) {
+        key = fmt::format("{}&continuation-token={}", key, *continuation_token);
+    }
```

Comment:
nit: the commit messages is framed as this change is unimportant, but I think the introduction of `start-after` and `continuation-token` should be highlighted, assuming they are functional changes

Reply:
Edited commit message.